### PR TITLE
Enable Falcon, which use MQA and therefore past_kv dimension is different and has its own if branch. 

### PR DESCRIPTION
**Pull request by qgao007**
## Type of Change

Enable Falcon model SQ. 
No API Change. 

## Description

Falcon can't run in previous default past_kv dimension settings, because it use Multi-Query Attention (MQA). Therefore, enabling Falcon using MQA key-value dimension is done. 

## Expected Behavior & Potential Risk

expected falcon model to run with sq. 

## Dependency Change?

No Dependency Change.

### CODE REVIEW

File: intel_extension_for_transformers/llm/evaluation/lm_eval/evaluator.py [hunk 1]
```diff
@@ -164,8 +164,8 @@ def evaluate(model,
         limit=limit,
         bootstrap_iters=bootstrap_iters,
         decontamination_ngrams_path=decontamination_ngrams_path,
-        write_out=write_out,
-        output_base_path=output_base_path
+        # write_out=write_out,
```
Comment (hshen14) [inline intel_extension_for_transformers/llm/evaluation/lm_eval/evaluator.py:167]: why comment out?

File: intel_extension_for_transformers/transformers/utils/utility.py [hunk 2]
```diff
@@ -93,6 +93,11 @@ def generate_dummy_past_key_values(input_bs, model):
         num_key_value_heads = normalized_config.num_key_value_heads
     if hasattr(normalized_config, "multi_query_group_num"):
         num_key_value_heads = normalized_config.multi_query_group_num
+    elif hasattr(normalized_config, "num_kv_heads"):
+        num_key_value_heads = normalized_config.num_kv_heads
+
```
Comment (hshen14) [inline intel_extension_for_transformers/transformers/utils/utility.py:98]: 3 space lines. pls format

File: intel_extension_for_transformers/llm/evaluation/lm_eval/evaluator.py (hunk 1 above)
Comment (qgao007) [reply intel_extension_for_transformers/llm/evaluation/lm_eval/evaluator.py:167]: Thanks for catching this. It's a version issues of lm-eval. Let me install the most recent version, validate and update this.

File: intel_extension_for_transformers/transformers/utils/utility.py (hunk 2 above)
Comment (qgao007) [reply intel_extension_for_transformers/transformers/utils/utility.py:98]: Thanks! will do.

File: intel_extension_for_transformers/llm/evaluation/lm_eval/evaluator.py (hunk 1 above)
Comment (qgao007) [reply intel_extension_for_transformers/llm/evaluation/lm_eval/evaluator.py]: validated lm-eval latest commit https://github.com/EleutherAI/lm-evaluation-harness.git@cc9778fbe4fa1a709be2abed9deb6180fd40e7e2.

**Pull request closed without merging**